### PR TITLE
feat: creating a blog by superadmin

### PR DESCRIPTION
**Pull request by zee-ham-su**
## What does the PR do?
This PR introduces a new endpoint for creating blog posts that is restricted to super admins. The endpoint only allows access if the provided token is valid and has the super admin role. The PR ensures that:

- [x] The endpoint handles missing or invalid authorization headers with a 401 status code
- [x] It returns a 403 status code if the token does not have the super admin role.
- [x] It correctly validates input fields and responds with a 400 status code for invalid requests.
- [x] On successful blog creation, it returns a 201 status code and the appropriate response body.

## How should this be manually tested?
1. Ensure you have a valid super admin token.
2. Send a POST request to the following endpoint:
POST /api/v1/blogs

Include the Authorization header:
 ```json

 Authorization: Bearer <access_token>
 ```
3. Provide a valid request body:
```
{
  "title": "Blog Title",
  "content": "<p>First paragraph.</p><p>Second paragraph.</p>",
  "tags": ["tag1", "tag2"],
  "image_urls": ["https://example.com/image1.jpg", "https://example.com/image2.jpg"]
}
```

### Success Response
on a successful execution, you will receive the following response structure:
blog_success

```
{
    {
    "status_code": 201,
    "message": "Blog created successfully",
    "data": {
        "blog_id": "unique-blog-id",
        "title": "Blog Title",
        "content": "<p>First paragraph.</p><p>Second paragraph.</p>",
        "image_urls": ["https://example.com/image1.jpg", "https://example.com/image2.jpg"],
        "tags": ["tag1", "tag2"],
        "author": "Super Admin",
        "created_at": "2024-08-05T19:03:52.194Z"
    }
}
}
```
### "Invalid Input" Response
If the request is invalid, you will receive the following response:
blog_invalidinput

```
{
    "status_code": 400,
    "message": "Invalid request. Title and content are required."
}


```
### Unauthorized Response
blog_unauthorized

```
{
    "status_code": 401,
    "message": "Unauthorized. Please log in."
}
```

### "Internal Server Error" Response

```
{
    "status_code": 500,
    "message": "Internal server error. Failed to create the blog post."
}
```

## Task
- [x]  Implemented POST /api/v1/blogs endpoint in blog.controller.ts.
- [x] Added method in BlogService to handle blog creation with super admin restriction.
- [x] Validated token and user role to ensure only super admins can create blogs.
- [x] Implemented appropriate error handling for invalid tokens and request bodies.. 
- [x] Added unit tests for the new endpoint covering authorization, input validation, and success cases
- [x] Updated API documentation to reflect the new endpoint and its requirements.

##Unit Test Evidence
blog_test

Reference Issue: [Link to Issue](https://linear.app/team-bingo/issue/BAC2-126/endpoint-for-creating-a-blog-or-46)

### CODE REVIEW

File: src/modules/blogs/entities/blog.entity.ts
Comment (Heba-WebDev) [inline src/modules/blogs/entities/blog.entity.ts:1]: Please specify the nullability of all columns

File: src/modules/blogs/dtos/create-blog.dto.ts
Comment (Heba-WebDev) [inline src/modules/blogs/dtos/create-blog.dto.ts:1]: Please add a minimum for the title length

File: src/modules/blogs/blog.service.ts
Comment (Heba-WebDev) [inline src/modules/blogs/blog.service.ts:1]: Please change the error thrown when a user is not found, with not found exception

File: src/modules/blogs/entities/blog.entity.ts
Comment (zee-ham-su) [reply src/modules/blogs/entities/blog.entity.ts:1]: done

File: src/modules/blogs/dtos/create-blog.dto.ts
Comment (zee-ham-su) [reply src/modules/blogs/dtos/create-blog.dto.ts:1]: added

File: src/modules/blogs/blog.service.ts
Comment (zee-ham-su) [reply src/modules/blogs/blog.service.ts:1]: done

File: src/modules/blogs/blog.controller.ts
Comment (ObodoakorDavid) [inline src/modules/blogs/blog.controller.ts:1]: Please change file to blogs.controller.ts to match the folder name, also add a status_code, message and data to the object you are returning. the newly created blog should be in the data. Also, move all logic to the service and call the service in the controller.

Comment (zee-ham-su) [reply src/modules/blogs/blog.controller.ts:1]: made this changes

File: src/modules/notifications/notifications.controller.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class NotificationsController {
     return this.notificationsService.markAllNotificationsAsReadForUser(userId);
   }
 
-  @Get()
+  @Get('/unread')
```
Comment (ObodoakorDavid) [inline src/modules/notifications/notifications.controller.ts:110]: Why are you changing the route for notifications?

Comment (zee-ham-su) [reply src/modules/notifications/notifications.controller.ts:110]: I worked on that in the previous stage and there's two get requests in the controller so one was overriding the other. So it was deployed, only one was showing and the other wasn't.  You can check it out yours self. Thanks

Comment (ObodoakorDavid) [reply src/modules/notifications/notifications.controller.ts:110]: Alright.

**Review by Heba-WebDev — APPROVED** [review]

**Review by Cobidex — APPROVED** [review]

**Review by CFCIfe — COMMENTED** [review]
LGTM

**Review by davidolanrewaju — APPROVED** [review]

**Pull request merged**